conn_test.go

```diff
@@ -0,0 +1,20 @@
+package amqp
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestConnProperty(t *testing.T) {
+	c := new(conn)
+	key := "propKey"
+	value := "propValue"
+	err := ConnProperty(key, value)(c)
+	if err != nil {
+		t.Error(fmt.Sprintf("%+v", err))
+	}
+
+	if c.properties[symbol(key)] != value {
+		t.Errorf("conn property was not set to key \"%s\" and value \"%s\"", key, value)
```

Comment:
Just FYI, the `%q` formatter is handy when you want to quote strings. It's particularly useful when comparing strings that may have unprintable characters. https://play.golang.org/p/xx2QI3_4Irr

Reply:
Oh, that's fancy. Thanks for spreading the knowledge.